include/lib/aarch32/smcc_macros.S

```diff
@@ -82,6 +121,44 @@
 
 	/* Restore the banked registers including the current SPSR */
 	add	r1, r0, #SMC_CTX_SP_USR
+
+#if ARM_ARCH_MAJOR == 7 && !ARCH_SUPPORTS_VIRTUALIZATION
```

Comment:
Please see my comment below regarding this. This can be changed to :
`#if ARM_ARCH_MAJOR == 7 && !defined(ARCH_SUPPORTS_VIRTUALIZATION)`